Upgrade to golang 1.6.2

runcom opened the pull request:
In preparation after #22000 is merged (so I can fix tests failures)

Signed-off-by: Antonio Murdaca runcom@redhat.com

runcom committed:
Upgrade to golang 1.6.2

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

thaJeztah commented:
I think you need to update more files than this one only, see https://github.com/docker/docker/pull/21978, and https://github.com/docker/docker/pull/21987.

Also the gccgo versions should probably be bumped then

runcom commented:
Yep, thanks @thaJeztah!

icecrime commented:
It's green :open_mouth:

runcom commented:
lol

thaJeztah commented:
@runcom will you be updating this PR with the other files to be changed as well?

runcom commented:
Yes

runcom commented:
@thaJeztah updated, have I updated everything now?

thaJeztah commented on Dockerfile:
```diff
@@ -127,19 +127,10 @@ RUN set -x \
-RUN curl -fsSL "https://storage.googleapis.com/golang/go${HACK_GO_VERSION}.linux-amd64.tar.gz" \
-	| tar -xzC /tmp \
-	&& mv /tmp/go "/usr/local/go${HACK_GO_VERSION}"
-
```
Are we sure this is no longer needed? Following this comment: https://github.com/golang/go/issues/15286#issuecomment-217323815 it looks like there's still a workaround needed (because a fix didn't make it for Go 1.6.2)

Also, if we're reverting the Hack/Workaround, it looks like there's also changes to be made to `hack/make/cross`; see https://github.com/docker/docker/pull/22082

ping @jstarks @jhowardmsft could you have a look?

thaJeztah replied:
Oh, and this note; [pkg/integration/utils_test.go#L32-L37](https://github.com/docker/docker/blob/5f2ba2b9ba5d1030dacb6c686d12c6a91c3c0612/pkg/integration/utils_test.go#L32-L37)

runcom replied:
took care of those :)

thaJeztah commented on Dockerfile.armhf:
```diff
@@ -65,9 +65,7 @@ RUN cd /usr/local/lvm2 \
-# are marked "end of life", see http://dave.cheney.net/unofficial-arm-tarballs
-ENV GO_VERSION 1.5.3
+ENV GO_VERSION 1.6.2
 RUN curl -fsSL "http://dave.cheney.net/paste/go${GO_VERSION}.linux-arm.tar.gz" \
```
This won't work; these builds are no longer updated, but, official builds should now be available :tada: see https://github.com/docker/docker/pull/21987#issue-147998486, and https://twitter.com/davecheney/status/720180308675403776

> With the release of Go 1.6 the Go team have made available official ARMv61 binary releases. Thus, this page is now end of life and will no longer be updated. The existing Go 1.4.x and 1.5.x releases will remain for a time to assist people bootstrapping ARM systems.
> For other arm variants, I recommend cross compiling or building from source:
> Build from source, http://dave.cheney.net/2015/09/04/building-go-1-5-on-the-raspberry-pi
> Cross compile, http://dave.cheney.net/2015/08/22/cross-compilation-with-go-1-5
> Notes
>
>ARMv6 is upwards compatible with ARMv7 systems. There is no material performance difference running an ARMv6 binary on an ARMv7 system.

I'm very bad at these, so not sure if these; https://storage.googleapis.com/golang/go1.6.2.linux-armv6l.tar.gz are compatible with armhf, or that we need to compile from source?

runcom replied:
good question - is there anyone we can ping for arm?

StefanScherer replied:
@runcom I'll have a look at it.

thaJeztah replied:
Thanks @StefanScherer! Wow this is becoming a group-effort-PR nice :heart:

thaJeztah commented:
Hm, some more things to take into account; 

The gccgo version has to be bumped (see the discussion on https://github.com/docker/docker/pull/22715) I already pinged @tophj-ibm @michael-holzheu there to help in getting those to the right version (and sha/hash)

Some things I'm not sure of; 

Looks like there's these patches; not sure if they need to be rebased for go 1.6.2 https://github.com/docker/docker/blob/af705ca0ae1af735661ae464cb53d93a890450e7/hack/vendor.sh#L66 (https://github.com/docker/go)

Oh, and I remember @tonistiigi was doing a patch somewhere to have https://go-review.googlesource.com/#/c/21654/, but not sure if that was applied somewhere in this repository or the runc or containerd repo @tonistiigi ?

thaJeztah commented:
Okay, for ppc64, gccgo and s390x, looks like we need to modify things
in these areas;


ppc64le:
[Dockerfile.ppc64le#L18](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.ppc64le#L18) (but this one is funny [Dockerfile.ppc64le#L90-L95](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.ppc64le#L90-L95))

gccgo:
[Dockerfile.gccgo#L9](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.gccgo#L9)

s390x:
[Dockerfile.s390x#L18](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.s390x#L18) (also [Dockerfile.s390x#L99-L104](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.s390x#L99-L104))

tophj-ibm commented:
@runcom should be safe to bump the ppc64le and gccgo dockerfiles to 6.1, there doesn't appear to be a 6.2 for gcc.

@thaJeztah for ppc64le, we have to manually make go which requires bootstrapping from go 1.4+. Because power wasn't supported in go 1.4, we have to do that from gccgo, hence the from gcc. I'm working on a pr to change this to xenial which has go for power available, but that's why that funny stuff is there 😛

runcom commented:
@tophj-ibm fixed ppc64le, not sure what there's to do in gccgo

tophj-ibm commented:
@runcom oh no, I meant bumping the base gcc images, because they include the gccgo versions.
 [Dockerfile.ppc64le#L18](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.ppc64le#L18) to ppc64le/gcc:6.1 and [Dockerfile.gccgo#L9](https://github.com/docker/docker/blob/cfb976438677b15ec023b2ad4822bf1b2f63cc8f/Dockerfile.gccgo#L9) to gcc:6.1

tophj-ibm commented on Dockerfile.ppc64le:
```diff
@@ -89,7 +89,7 @@ RUN set -x \
 ## BUILD GOLANG 1.6
 # NOTE: ppc64le has compatibility issues with older versions of go, so make sure the version >= 1.6
-ENV GO_VERSION 1.6.2
+ENV GO_VERSION 1.6.1
```
this is fine at 1.6.2

runcom replied:
fixed

tophj-ibm commented:
seems good to me, thanks!

thaJeztah commented:
Thanks @tophj-ibm!

Guess we need to await input from the Windows people now (w.r.t. the 1.5.3 hack)

jstarks commented:
Nano Server TP5 is incompatible with Go 1.6.2. We have to fix that incompatibility in Windows, so until there is another release we are stuck. We could move Windows to 1.6.1 though.

Another possible workaround just occurred to me though, so give me a few hours to try it out.

jstarks committed:
Windows: work around Go 1.6.2/Nano Server TP5 issue

This works around golang/go#15286 by explicitly loading shell32.dll at
load time, ensuring that syscall can load it dynamically during process
startup.

Signed-off-by: John Starks <jostarks@microsoft.com>
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

jstarks commented:
I found a workaround. I can submit it as a separate PR, or you can cherry pick jstarks/docker@b7a721a0351c35af5d9b57749cb776e6f99597fe as part of this.

icecrime commented:
Failure on ARM:

```
10:45:21 Step 7 : RUN curl -fsSL "http://dave.cheney.net/paste/go${GO_VERSION}.linux-arm.tar.gz" 	| tar -xzC /usr/local
10:45:21  ---> Running in cb6ca9f004db
10:45:22 curl: (22) The requested URL returned error: 404 Not Found
```

tonistiigi commented:
@thaJeztah The patch we use is for the `golang/net` package not for the main one.

StefanScherer commented:
@runcom this PR doesn't update the `contrib/builder/deb/armhf/debian-jessie/Dockerfile` yet  https://github.com/runcom/docker/blob/a06091c680809cdcc70c492691c8d0602e6040f7/contrib/builder/deb/armhf/debian-jessie/Dockerfile#L9-L10

Until now we took Go 1.4.3 to build docker for ARM as of https://github.com/docker/docker/issues/18125
But as this sounds good I suggest to update both Dockerfiles to 1.6.2 and I'll check the deb package as well.

Currently running an integration test with golang 1.6.2 at https://jenkins.hypriot.com/job/docker%20pr22840/2/console
where I fixed the golang download url in `Dockerfile.armhf` from

http://dave.cheney.net/paste/go${GO_VERSION}.linux-arm.tar.gz

to

https://storage.googleapis.com/golang/go${GO_VERSION}.linux-armv6l.tar.gz

alexellis commented:
@StefanScherer I've had mixed experiences building Docker in Docker on ARM. Arch Linux is building direct on the machine against Go 1.6. I can try and build this on Arch Linux ARM v6 / v7 once you've updated the ARM Dockerfile?

thaJeztah commented:
@icecrime yes, ARM failure is due to https://github.com/docker/docker/pull/22840#discussion_r64018444

StefanScherer committed:
Update golang 1.6.2 for ARM

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

StefanScherer commented:
@runcom You may cherry-pick https://github.com/hypriot/docker/commit/9635ec6e8d3d325c71058ec35bf544154563b19b or do it yourself in the PR to update both ARM Dockerfiles to golang 1.6.2. Thanks.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "go1.6" git@github.com:runcom/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~3
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

runcom commented:
I've cherry-picked @StefanScherer and @jstarks commits here - anything else left?

StefanScherer commented:
@thaJeztah Can you restart the arm build please?

runcom commented:
@StefanScherer I've started an arm build

LK4D4 commented:
@runcom both arm and experimental has similar failures :/

StefanScherer commented:
Yes, arm build finishes with `OOPS: 1256 passed, 55 skipped, 5 FAILED`. We've started monitoring integrations tests in the last days again for master branch and see about that number of failing tests before this PR. It seems we need some fixing/skipping work for ARM, but that should go into other PR's.

The problems of #18125 really seem to be solved, we had about 50% failing integration tests in November with Go 1.5. We have also tested the 1.11.1 with golang 1.6.2 on RPi's and haven't seen problems.

alexellis commented:
@StefanScherer out of curiosity, where are the arm build hosted that are listed on Jenkins? How long are your PI2/3 builds taking?

To try this do I just need to clone runcom:go1.6?

StefanScherer commented:
@alexellis Our Jenkins slaves are Scaleway C1 machines (ARMv7), there is also a docker-machine plugin for these. Yes, the PR is now sufficient, either clone the forked repo or have a look at GitHub docs [checkout out pull requests locally](https://help.github.com/articles/checking-out-pull-requests-locally/)

runcom commented:
@jstarks windowsTP5 also has a panic in unit tests

runcom commented:
@LK4D4 I've restarted experimental, @StefanScherer so arm is fine with those failures?

runcom commented:
@LK4D4 for some reason the error in experimental is `no space left on device` :confused:

StefanScherer commented:
@runcom Yes the remaining arm failures are not related to this PR.

cpuguy83 commented:
```
19:14:48 --- FAIL: TestIsKilledTrueWithKilledProcess (0.01s)
19:14:48 panic: runtime error: invalid memory address or nil pointer dereference [recovered]
19:14:48 	panic: runtime error: invalid memory address or nil pointer dereference
19:14:48 [signal 0xc0000005 code=0x1 addr=0x0 pc=0x4615d9]
19:14:48 
19:14:48 goroutine 84 [running]:
19:14:48 panic(0x674520, 0xc08200a080)
19:14:48 	c:/go/src/runtime/panic.go:481 +0x3f4
19:14:48 testing.tRunner.func1(0xc0821483f0)
19:14:48 	c:/go/src/testing/testing.go:467 +0x199
19:14:48 panic(0x674520, 0xc08200a080)
19:14:48 	c:/go/src/runtime/panic.go:443 +0x4f7
19:14:48 os.(*Process).signal(0x0, 0x8d0078, 0xc08200a1d0, 0x0, 0x0)
19:14:48 	c:/go/src/os/exec_windows.go:58 +0x289
19:14:48 os.(*Process).Signal(0x0, 0x8d0078, 0xc08200a1d0, 0x0, 0x0)
19:14:48 	c:/go/src/os/doc.go:55 +0x48
19:14:48 os.(*Process).kill(0x0, 0x0, 0x0)
19:14:48 	c:/go/src/os/exec_posix.go:53 +0x4c
19:14:48 os.(*Process).Kill(0x0, 0x0, 0x0)
19:14:48 	c:/go/src/os/doc.go:40 +0x34
19:14:48 github.com/docker/docker/pkg/integration.TestIsKilledTrueWithKilledProcess(0xc0821483f0)
19:14:48 	c:/go/src/github.com/docker/docker/pkg/integration/utils_test.go:41 +0xba
19:14:48 testing.tRunner(0xc0821483f0, 0x8125a8)
19:14:48 	c:/go/src/testing/testing.go:473 +0x9f
19:14:48 created by testing.RunTests
```

runcom commented:
@cpuguy83 already pinged @jstarks https://github.com/docker/docker/pull/22840#issuecomment-220914210 probably @jhowardmsft also

jstarks commented:
@runcom I think it was a mistake to remove the block for that test. You can either put the block back, or you can cherry pick jstarks/docker@5391528869a1bf308d014615b2f8fa4c57be5aff, which ports that test and the one above it to work on Windows.

runcom commented:
@jstarks thx, I've put that block back (better to have another PR to port those tests)

michael-holzheu commented:
I tried to build Dockerfile.s390x with "FROM s390x/gcc:6.1" on s390x.
The notary compile failed with:

```
Step 19 : RUN set -x    && export GO15VENDOREXPERIMENT=1        && export GOPATH="$(mktemp -d)"         && git clone https://github.com/docker/notary.git "$GOPATH/src/github.com/docker/notary"        && (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION" && ln -s . vendor/src)       && GOPATH="$GOPATH/src/github.com/docker/notary/vendor:$GOPATH"                 go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server     && GOPATH="$GOPATH/src/github.com/docker/notary/vendor:$GOPATH"                 go build -o /usr/local/bin/notary github.com/docker/notary/cmd/notary   && rm -rf "$GOPATH"
 ---> Running in 2681ce9cbc5b
+ export GO15VENDOREXPERIMENT=1
+ mktemp -d                                                                                                                                     
+ export GOPATH=/tmp/tmp.utEIc6G4bp                                                                                                             
+ git clone https://github.com/docker/notary.git /tmp/tmp.utEIc6G4bp/src/github.com/docker/notary                                               
Cloning into '/tmp/tmp.utEIc6G4bp/src/github.com/docker/notary'...                                                                              
+ cd /tmp/tmp.utEIc6G4bp/src/github.com/docker/notary                                                                                           
+ git checkout -q v0.3.0                                                                                                                        
+ ln -s . vendor/src                                                                                                                            
+ GOPATH=/tmp/tmp.utEIc6G4bp/src/github.com/docker/notary/vendor:/tmp/tmp.utEIc6G4bp go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server                                                                                                                        
# github.com/docker/notary/vendor/github.com/gorilla/mux                                                                                        
tmp/tmp.utEIc6G4bp/src/github.com/docker/notary/vendor/github.com/gorilla/mux/mux.go:12:28: error: import file 'github.com/gorilla/context' not found                                                                                                                                           
  "github.com/gorilla/context"

tmp/tmp.Mw4OatwWlq/src/github.com/docker/notary/vendor/github.com/gorilla/mux/mux.go:96:9: error: reference to undefined name 'context'         
   defer context.Clear(req)
...
```

thaJeztah commented:
@runcom @amitkris I guess this also needs to be updated as part of this PR; https://github.com/docker/docker/blob/master/hack/make.sh#L117-L121

amitkris commented:
@thaJeztah thanks for the heads up!
@runcom could you please change https://github.com/docker/docker/blob/master/hack/make.sh#L120
to 
 export GOPATH="${GOPATH}:/usr/lib/gocode/1.6.2" 
as a part of this PR?

runcom commented:
@amitkris sure I can

runcom commented:
@amitkris done - @thaJeztah @michael-holzheu not sure what's going on with notary - who can I ping for it?

thaJeztah commented:
ping @endophage @cyli can you have a look for the Notary problems?

cyli commented:
`github.com/gorilla/context` has been vendored into notary, so it should be there.  I'm wondering if maybe the `ln -s` may be causing issues since it's actually expecting the go 1.5+ vendoring now?  Does it still fail if you remove the `ln -s`?  

Apologies, I would just test myself, but I can't seem to even build the s390x Dockerfile.  I'm not sure why, but I'm getting:

```
...
Status: Downloaded newer image for s390x/gcc:6.1
 ---> a5f2322042ba
Step 2 : RUN apt-get update && apt-get install -y 	apparmor 	aufs-tools 	automake 	bash-completion 	btrfs-toolsbuild-essential 	createrepo 	curl 	dpkg-sig 	git 	iptables 	jq 	net-tools 	libapparmor-dev 	libcap-dev libltdl-dev 	libsqlite3-dev 	libsystemd-journal-dev 	libtool 	mercurial 	pkg-config 	python-dev 	python-mock 	python-pip 	python-websocket 	xfsprogs 	tar 	--no-install-recommends
 ---> Running in 0211be813cb5
Container command '/bin/sh' not found or does not exist.
```

thaJeztah commented:
ping @runcom is there anything left to be done in this PR? Do we need to revert changes for the Windows Hack, or is that taken care of?

thaJeztah commented:
triggered ARM and PowerPC as well, they were not running on this PR

runcom commented:
I think we're done here afaict, I've already reverted the windows check so it should be good to go

thaJeztah commented:
@runcom cool. the s390x is going in a follow up?

runcom commented:
I'm not sure what's going on there honestly - and I cannot even test it. If anyone is willing to point me in the right direction I can make the modifications needed here, otherwise, yes, I'll defer that to another PR since I have to way to try that out

thaJeztah commented:
@runcom sgtm; let's see if ARM and PowerPC go green, then I think we're ready :tada:

tophj-ibm commented:
Probably safe to leave z out of this one, I know that team is testing out a gc build anyway.

runcom commented:
This is so green :+1:

thaJeztah commented:
Just thought of this; should we remove the `GO15VENDOREXPERIMENT` in the Dockerfiles as part of this PR?
https://github.com/docker/docker/search?utf8=✓&q=GO15VENDOREXPERIMENT

LGTM otherwise

LK4D4 commented:
@thaJeztah let's do it in separate PR
LGTM

michael-holzheu commented:
@cyli 

> @runcom @thaJeztah github.com/gorilla/context has been vendored into notary, so it should be there. I'm wondering if maybe the ln -s may be causing issues since it's actually expecting the go 1.5+ vendoring now? Does it still fail if you remove the ln -s?

Looks like the softlink trick does no longer work for gcc 6.1. I also expected that with gcc 6.1 / go 1.6.1 we can remove the softlink workaround, but when I remove the softlink the build still fails. I assume gccgo has problems with vendoring? Perhaps golang/go#15814 is related?

I could compile notary when I keep the softlink and explicitly disable GO15VENDOREXPERIMENT:
```
diff --git a/Dockerfile.s390x b/Dockerfile.s390x
index bb7ec8b..4b8edd4 100644
--- a/Dockerfile.s390x
+++ b/Dockerfile.s390x
@@ -15,7 +15,7 @@
 # the case. Therefore, you don't have to disable it anymore.
 #
 
-FROM s390x/gcc:5.3
+FROM s390x/gcc:6.1
 
 # Packaged dependencies
 RUN apt-get update && apt-get install -y \
@@ -132,7 +132,7 @@ RUN set -x \
 # Install notary and notary-server
 ENV NOTARY_VERSION v0.3.0
 RUN set -x \
-       && export GO15VENDOREXPERIMENT=1 \
+       && export GO15VENDOREXPERIMENT=0 \
        && export GOPATH="$(mktemp -d)" \
        && git clone https://github.com/docker/notary.git "$GOPATH/src/github.com/docker/notary" \
        && (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION" && ln -s . vendor/src) \
@@ -158,7 +158,7 @@ RUN useradd --create-home --gid docker unprivilegeduser
```

To summarize:

| GO15VENDOREXPERIMENT | Softlink | Success |
|----------------------|----------|---------|
| 1                    | yes      | no      |
| 1                    | no       | no      |
| unset                | yes      | no      |
| unset                | no       | no      |
| 0                    | yes      | yes     |
| 0                    | no       | no      |

What about Dockerfile.gcc on x86. Does this work with gcc 6.1?

> Apologies, I would just test myself, but I can't seem to even build the s390x Dockerfile. I'm not sure why, but I'm getting:

Do you have a s390x machine for testing?

michael-holzheu commented:
> What about Dockerfile.gccgo on x86. Does this work with gcc 6.1?

Ok: Dockerfile.gccgo does not compile notary.

cyli commented:
Thanks for figuring this out!  I would have thought censoring would work
too.

And no I do not - I guess that would be why.  :). I figured I'd originally
thought I'd be able to build but not run.

On Monday, May 30, 2016, Michael Holzheu <notifications@github.com> wrote:

> @cyli <https://github.com/cyli>
>
> @runcom <https://github.com/runcom> @thaJeztah
> <https://github.com/thaJeztah> github.com/gorilla/context has been
> vendored into notary, so it should be there. I'm wondering if maybe the ln
> -s may be causing issues since it's actually expecting the go 1.5+
> vendoring now? Does it still fail if you remove the ln -s?
>
> Looks like the softlink trick does no longer work for gcc 6.1. I also
> expected that with gcc 6.1 / go 1.6.1 we can remove the softlink
> workaround, but when I remove the softlink the build still fails. I assume
> gccgo has problems with vendoring? Perhaps golang/go#15814
> <https://github.com/golang/go/issues/15814> is related?
>
> I could compile notary when I keep the softlink and explicitly disable
> GO15VENDOREXPERIMENT:
>
> diff --git a/Dockerfile.s390x b/Dockerfile.s390x
> index bb7ec8b..4b8edd4 100644
> --- a/Dockerfile.s390x
> +++ b/Dockerfile.s390x
> @@ -15,7 +15,7 @@
>  # the case. Therefore, you don't have to disable it anymore.
>  #
>
> -FROM s390x/gcc:5.3
> +FROM s390x/gcc:6.1
>
>  # Packaged dependencies
>  RUN apt-get update && apt-get install -y \
> @@ -132,7 +132,7 @@ RUN set -x \
>  # Install notary and notary-server
>  ENV NOTARY_VERSION v0.3.0
>  RUN set -x \
> -       && export GO15VENDOREXPERIMENT=1 \
> +       && export GO15VENDOREXPERIMENT=0 \
>         && export GOPATH="$(mktemp -d)" \
>         && git clone https://github.com/docker/notary.git "$GOPATH/src/github.com/docker/notary" \
>         && (cd "$GOPATH/src/github.com/docker/notary" && git checkout -q "$NOTARY_VERSION" && ln -s . vendor/src) \
> @@ -158,7 +158,7 @@ RUN useradd --create-home --gid docker unprivilegeduser
>
> To summarize:
> GO15VENDOREXPERIMENT Softlink Success
> 1 yes no
> 1 no no
> unset yes no
> unset no no
> 0 yes yes
> 0 no no
>
> What about Dockerfile.gcc on x86. Does this work with gcc 6.1?
>
> Apologies, I would just test myself, but I can't seem to even build the
> s390x Dockerfile. I'm not sure why, but I'm getting:
>
> Do you have a s390x machine for testing?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/docker/pull/22840#issuecomment-222487942>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AANn4BlhRQKSgioRjDya6C8pHZ0PmkYhks5qGuAQgaJpZM4IilDZ>
> .
>

michael-holzheu commented:
So should we update Dockerfile.s390x (with the GO15VENDOREXPERIMENT=0 workaround) and Dockerfile.gccgo to gcc6.1 to be compliant to your go 1.6 build requirement?

cyli commented:
@michael-holzheu That would be necessary if you wanted yubikey integrations in those two distributions :)  Thank you!
